components/board/customlinux/setup.go

```diff
@@ -0,0 +1,83 @@
+// Package customlinux implements a board running linux
+package customlinux
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/pkg/errors"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/components/board/genericlinux"
+)
+
+// GenericLinuxPin describes a gpio pin on a linux board.
+type GenericLinuxPin struct {
+	Name            string `json:"name"`
```

Comment:
this structure makes more sense for parsing the json format where they input a list of pins, not a map.

Reply:
since it is no longer a pointer to  a config too - please test it out live to see if you get nil dereference errors once our config and reader logic clears the attributes after consuming them. 

This can be tested live by changing a ton of attributes or writing a test that makes use of `config.ProcessConfig` to make sure all the attributes are turned into ConvertedAttributes.

Reply:
note the function isn't exported so you'll have to setup the harness for the parent function to test that.